core/controllers/classifier.py

```diff
@@ -123,3 +129,61 @@ def post(self):
         classifier_services.mark_training_job_complete(job_id)
 
         return self.render_json({})
+
+
+def update_failed_jobs(job_models):
+    for job_model in job_models:
+        classifier_services.mark_training_job_failed(job_model.id)
+
+
+def fetch_training_data(exp_id, state):
```

Comment:
@pranavsid98  Invalidating older jobs and creating new jobs has an issue of starvation. When you invalidate old job the new job will be pushed back and now it will be trained much later. But if you change older job then we can tackle the issue of starvation.